### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by Ningsir** [quiver-team/torch-quiver#89]
The memory allocating by cudaMalloc in [quiver_sample](https://github.com/quiver-team/torch-quiver/blob/main/srcs/cpp/src/quiver/cuda/quiver_sample.cu#L391-L391) does not seem to be freed.
I think it can be freed in [destructor function](https://github.com/quiver-team/torch-quiver/blob/main/srcs/cpp/include/quiver/quiver.cu.hpp#L279-L279).

**Comment by eedalong** [comment]
Hi, thx!  We did notice this problem and I also tried to do memory management and object delete logic before open-source in this [branch](https://github.com/quiver-team/torch-quiver/tree/feature/quiver_del). Actually It's a pretty complicated problem because we need to take care about memory lifetime management in DDP training. It's indeed an important problem and we will fix it in the future. But fortunately，for now this will not affect your training because these memory lives until training is finished
